File: src/main/java/dev/jolvera/finjav/models/Recurrence.java
package dev.jolvera.finjav.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Recurrence {
    private UUID id;
    private UUID userId;
    private LocalDate dateCreated;
    private LocalDate dateModified;
    private String name;
    private Boolean isIncome;
    private LocalDate recurringDate;
}

File: src/main/java/dev/jolvera/finjav/models/Recurrence.kt
package dev.jolvera.finjav.models

import java.math.BigDecimal
import java.time.LocalDate
import java.util.UUID

data class Recurrence(
    val id: UUID?,
    val userId: UUID?,
    val dateCreated: LocalDate,
    val dateModified: LocalDate,
    val startDate: LocalDate,
    val name: String,
    val amount: BigDecimal,
    val isIncome: Boolean,
    val recurringDate: LocalDate,
)
